src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb

```diff
@@ -709,22 +709,28 @@ Namespace Microsoft.VisualStudio.Editors.AppDesCommon
 
         Public Class SerializationProvider
 
+            Private Shared ReadOnly KnownTypes As Type() = {GetType(Size)}
+
             Public Shared Sub Serialize(stream As Stream, value As Object)
-                If value Is Nothing Then
+                If stream Is Nothing Or value Is Nothing Then
                     Return
                 End If
                 Using writer As New BinaryWriter(stream, Encoding.UTF8, True)
                     Dim valueType = value.GetType()
                     writer.Write(valueType.AssemblyQualifiedName)
                     writer.Flush()
-                    Call New DataContractSerializer(valueType).WriteObject(stream, value)
+                    Call New DataContractSerializer(valueType, KnownTypes).WriteObject(stream, value)
                 End Using
             End Sub
 
             Public Shared Function Deserialize(stream As Stream) As Object
+                If stream Is Nothing OrElse stream.Length = 0 Then
+                    Return Nothing
+                End If
                 Using reader As New BinaryReader(stream, Encoding.UTF8, True)
+                    reader.BaseStream.Position = 0
```

Comment:
A deserialize function should not assume it is reading from the start of the stream. If you need to reset the position in test code, do that in the test itself.

Reply:
Removed this line.